packages/grid/src/vaadin-grid-selection-column.js

```diff
@@ -149,6 +162,132 @@ class GridSelectionColumn extends GridColumn {
     checkbox.indeterminate = this.__indeterminate;
   }
 
+  /** @private */
+  __lassoAutoScroller() {
+    if (this.__lassoDragStartIndex !== undefined) {
+      // Get the row being hovered over
+      const renderedRows = this._grid._getRenderedRows();
+      const hoveredRow = renderedRows.find((row) => {
+        const rowRect = row.getBoundingClientRect();
+        return this.__lassoCurrentY >= rowRect.top && this.__lassoCurrentY <= rowRect.bottom;
+      });
+
+      // Get the index of the row being hovered over or the first/last
+      // visible row if hovering outside the grid
+      let hoveredIndex = hoveredRow ? hoveredRow.index : undefined;
+      const scrollableArea = this.__getScrollableArea();
+      if (this.__lassoCurrentY < scrollableArea.top) {
+        hoveredIndex = this._grid._firstVisibleIndex;
+      } else if (this.__lassoCurrentY > scrollableArea.bottom) {
+        hoveredIndex = this._grid._lastVisibleIndex;
+      }
+
+      if (hoveredIndex !== undefined) {
+        // Select all items between the start and the current row
+        renderedRows.forEach((row) => {
+          if (
+            (hoveredIndex > this.__lassoDragStartIndex &&
+              row.index >= this.__lassoDragStartIndex &&
+              row.index <= hoveredIndex) ||
+            (hoveredIndex < this.__lassoDragStartIndex &&
+              row.index <= this.__lassoDragStartIndex &&
+              row.index >= hoveredIndex)
+          ) {
+            if (this.__lassoSelect) {
+              this._grid.selectItem(row._item);
+            } else {
+              this._grid.deselectItem(row._item);
+            }
+          }
+        });
+      }
+
+      // Start scrolling in the top/bottom 15% of the scrollable area
+      const scrollTriggerArea = scrollableArea.height * 0.15;
+      // Maximum number of pixels to scroll per iteration
+      const maxScrollAmount = 10;
+
+      if (this.__lassoDy < 0 && this.__lassoCurrentY < scrollableArea.top + scrollTriggerArea) {
+        const dy = scrollableArea.top + scrollTriggerArea - this.__lassoCurrentY;
+        const percentage = Math.min(1, dy / scrollTriggerArea);
+        this._grid.$.table.scrollTop -= percentage * maxScrollAmount;
+      }
+      if (this.__lassoDy > 0 && this.__lassoCurrentY > scrollableArea.bottom - scrollTriggerArea) {
+        const dy = this.__lassoCurrentY - (scrollableArea.bottom - scrollTriggerArea);
+        const percentage = Math.min(1, dy / scrollTriggerArea);
+        this._grid.$.table.scrollTop += percentage * maxScrollAmount;
+      }
```

Comment:
Scroll logic is lacking tests. Since this could get tricky with specific timing / scroll values, I would suggest to start with simple tests that checks that scrolling upwards / downwards happens when dragging in the top / bottom regions of the grid body, but not within the center of it.

Reply:
In order to check the scrolling upwards / downwards works, should assertions test this._grid.$.table.scrollTop value?

Reply:
Yes, we could start with tests that simply check if `scrollTop` is increased / decreased or no modified.

Reply:
I've added a couple of tests.

Reply:
@flang I've rebased your PR to include a fix for a test helper (https://github.com/vaadin/web-components/pull/6223), and added a commit for simplifying the tests a bit based on that.